File: example_code/fizzbuzz.cpp
template <typename T>
static void fizzbuzz(std::ostream& out, const T& number, const std::vector<std::pair<std::string, T>>& values)
{
    bool found = false;
 
    std::for_each(values.begin(), values.end(),
        [&](decltype(*values.begin()) i)
        {
            if (0 == number % i.second)
            {
                out << i.first.c_str();
                found = true;
            }        
        }
    );
 
    if (!found)
    {
        out << number;
    }
}

File: example_code/fizzbuzz.js
console.log(
    Array.apply(null, { length: 100 })
         .map(Number.call, Number)
         .map(function(n) {
        return n % 15 === 0 ?
                  'FizzBuzz' : n % 3 === 0 ?
                  'Fizz' : n % 5 === 0 ? 'Buzz' : n;
    })
);